pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/AbstractNode.java

```diff
@@ -28,25 +29,52 @@
 
 /**
  * Base class for all implementations of the Node interface.
+ *
+ * <p>Please use the {@link Node} interface wherever possible and
+ * not this class, unless you're compelled to do so.
+ *
+ * <p>Note that nearly all methods of the {@link Node} interface
+ * will have default implementations with PMD 7.0.0, so that it
+ * will not be necessary to extend this class directly.
  */
 public abstract class AbstractNode implements Node {
 
     private static final Logger LOG = Logger.getLogger(AbstractNode.class.getName());
 
+    /**
+     * @deprecated Use {@link #getParent()}
+     */
+    @Deprecated
     protected Node parent;
+    @Deprecated
     protected Node[] children;
+    /** @deprecated Use {@link #getIndexInParent()} */
+    @Deprecated
     protected int childIndex;
+    /** @deprecated Use {@link #jjtGetId()} if you are a jjtree node. */
+    @Deprecated
     protected int id;
-
-    private String image;
+    /** @deprecated This will be removed to delegate to the tokens for nodes that are backed by tokens. */
+    @Deprecated
     protected int beginLine = -1;
+    /** @deprecated This will be removed to delegate to the tokens for nodes that are backed by tokens. */
+    @Deprecated
     protected int endLine;
+    /** @deprecated This will be removed to delegate to the tokens for nodes that are backed by tokens. */
+    @Deprecated
     protected int beginColumn = -1;
+    /** @deprecated This will be removed to delegate to the tokens for nodes that are backed by tokens. */
+    @Deprecated
     protected int endColumn;
-    private DataFlowNode dataFlowNode;
-    private Object userData;
+    // Those should have been private.
+    @Deprecated
     protected GenericToken firstToken;
+    @Deprecated
     protected GenericToken lastToken;
+    private DataFlowNode dataFlowNode;
+    private Object userData;
+    // @Deprecated?
```

Comment:
I think, it doesn't matter much atm whether "image" is deprecated or not - here it is private anyways.